python-checks/src/main/java/org/sonar/python/checks/IncorrectExceptionTypeCheck.java

```diff
@@ -42,26 +43,27 @@ public void initialize(Context context) {
       if (raiseStatement.expressions().isEmpty()) {
         return;
       }
-      if (raiseStatement.expressions().get(0).is(Tree.Kind.CALL_EXPR)) {
-        CallExpression callExpression = ((CallExpression) raiseStatement.expressions().get(0));
+      Expression raisedExpression = raiseStatement.expressions().get(0);
+      if (raisedExpression.is(Tree.Kind.CALL_EXPR)) {
+        CallExpression callExpression = ((CallExpression) raisedExpression);
         Symbol calleeSymbol = callExpression.calleeSymbol();
-        if (!inheritsFromBaseException(calleeSymbol)) {
+        if (!inheritsFromBaseException(calleeSymbol, raisedExpression)) {
           ctx.addIssue(raiseStatement, MESSAGE);
         }
       }
-      if (raiseStatement.expressions().get(0).is(Tree.Kind.NAME)) {
-        Symbol symbol = ((Name) raiseStatement.expressions().get(0)).symbol();
-        if (!inheritsFromBaseException(symbol)) {
+      if (raisedExpression.is(Tree.Kind.NAME)) {
+        Symbol symbol = ((Name) raisedExpression).symbol();
+        if (!inheritsFromBaseException(symbol, raisedExpression)) {
           ctx.addIssue(raiseStatement, MESSAGE);
         }
       }
-      if (raiseStatement.expressions().get(0).is(Tree.Kind.STRING_LITERAL)) {
+      if (raisedExpression.is(Tree.Kind.STRING_LITERAL)) {
```

Comment:
As discussed, unfortunately this is not possible (yet) because we don't have a precise modelization of types imported from typeshed: "str" class symbol has an unresolved type hierarchy.
I suggest to keep checking the tree kind for the time being, and come back to this implementation once we will improve typeshed support